aws_lambda_builders/workflows/nodejs_npm_esbuild/workflow.py

```diff
@@ -53,150 +50,75 @@ def __init__(self, source_dir, artifacts_dir, scratch_dir, manifest_path, runtim
-        )
-
-    def actions_with_bundler(
-        self, source_dir, scratch_dir, artifacts_dir, bundler_config, osutils, subprocess_npm, subprocess_esbuild
-    ) -> List[BaseAction]:
-        """
-        Generate a list of Nodejs build actions with a bundler
-
-        :type source_dir: str
-        :param source_dir: an existing (readable) directory containing source files
-
-        :type scratch_dir: str
-        :param scratch_dir: an existing (writable) directory for temporary files
-
-        :type artifacts_dir: str
-        :param artifacts_dir: an existing (writable) directory where to store the output.
-
-        :type bundler_config: dict
-        :param bundler_config: configurations for the bundler action
-
-        :type osutils: aws_lambda_builders.workflows.nodejs_npm.utils.OSUtils
-        :param osutils: An instance of OS Utilities for file manipulation
-
-        :type subprocess_npm: aws_lambda_builders.workflows.nodejs_npm.npm.SubprocessNpm
-        :param subprocess_npm: An instance of the NPM process wrapper
-
-        :type subprocess_esbuild: aws_lambda_builders.workflows.nodejs_npm_esbuild.esbuild.SubprocessEsbuild
-        :param subprocess_esbuild: An instance of the esbuild process wrapper
-
-        :rtype: list
-        :return: List of build actions to execute
-        """
-        actions: List[BaseAction] = [
-            CopySourceAction(source_dir, scratch_dir, excludes=self.EXCLUDED_FILES + tuple(["node_modules"]))
-        ]
-
-        # Bundle dependencies separately in a dependency layer. We need to check the esbuild
-        # version here to ensure that it supports skipping dependency bundling
-        esbuild_no_deps = [
-            EsbuildCheckVersionAction(scratch_dir, subprocess_esbuild),
-            EsbuildBundleAction(
-                scratch_dir,
-                artifacts_dir,
-                bundler_config,
-                osutils,
-                subprocess_esbuild,
-                self.manifest_path,
-                skip_deps=True,
-            ),
-        ]
-        esbuild_with_deps = EsbuildBundleAction(
-            scratch_dir, artifacts_dir, bundler_config, osutils, subprocess_esbuild, self.manifest_path
-        )
+        if not self.download_dependencies and not self.dependencies_dir:
+            # Invalid workflow, can't have no dependency dir and no installation
+            raise EsbuildExecutionError(
+                message="Lambda Builders encountered an invalid workflow. A workflow can't "
+                "include a dependencies directory without installing dependencies."
+            )
 
-        install_action = NodejsNpmWorkflow.get_install_action(
-            source_dir,
-            scratch_dir,
-            subprocess_npm,
-            osutils,
-            self.options,
+        self.actions.append(
+            CopySourceAction(source_dir=self.source_dir, dest_dir=self.scratch_dir, excludes=self.EXCLUDED_FILES)
```

Comment:
I think using named keywords can be more readable in cases where it's not immediately obvious what the parameter(s) would be for. E.g. here we would immediately know which folder we copy from and which one we copy to, without having to check the CopySourceAction definition.